### PR TITLE
Reduce unnecessary log noise

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Reduce unnecessary log noise in redirection handler

**Why?**
admin handler & other future handler does not need redirection, reduce unnecessary noise

**How did you test it?**
N/A

**Potential risks**
N/A

**Is hotfix candidate?**
N/A

### CODE REVIEW

File: service/frontend/redirection_interceptor.go [hunk 1]
```diff
@@ -177,8 +177,9 @@ func (i *RedirectionInterceptor) Intercept(
 		return i.handleRedirectAPIInvocation(ctx, req, info, handler, methodName, respAllocFn, namespaceName)
 	}
 
-	// this should not happen, log warn and move on
-	i.logger.Warn(fmt.Sprintf("RedirectionInterceptor encountered unknown API: %v", methodName))
+	// this should not happen for frontend APIs but ok for admin and other future handlers
+	// log warn and move on
```
Comment (yux0) [inline service/frontend/redirection_interceptor.go:181]: nit: update comment

**Review by yux0 — APPROVED** [review]

File: service/frontend/redirection_interceptor.go [hunk 2]
```diff
@@ -177,8 +177,9 @@ func (i *RedirectionInterceptor) Intercept(
 		return i.handleRedirectAPIInvocation(ctx, req, info, handler, methodName, respAllocFn, namespaceName)
 	}
 
-	// this should not happen, log warn and move on
-	i.logger.Warn(fmt.Sprintf("RedirectionInterceptor encountered unknown API: %v", methodName))
+	// this should not happen for frontend APIs but ok for admin and other future handlers
```
Comment (dnr) [inline service/frontend/redirection_interceptor.go:180]: You can get the service name from info.FullMethod and log at Warn level for WorkflowService, and Debug level for AdminService or others. That might be more useful to catch bugs

File: service/frontend/redirection_interceptor.go [hunk 3]
```diff
@@ -177,8 +177,9 @@ func (i *RedirectionInterceptor) Intercept(
...
+	// log warn and move on
+	i.logger.Debug(fmt.Sprintf("RedirectionInterceptor encountered unknown API: %v", methodName))
```
Comment (dnr) [inline service/frontend/redirection_interceptor.go:182]:
```suggestion
	i.logger.Debug("RedirectionInterceptor encountered unknown API", tag.Name(info.FullMethod))
```
I'd rather see the full name to know what service it's in

File: service/frontend/redirection_interceptor.go (hunk 2 above)
Comment (wxing1292) [reply service/frontend/redirection_interceptor.go:180]: https://github.com/temporalio/temporal/blob/master/service/frontend/redirection_interceptor_test.go#L183

all workflow service APIs are validated

File: service/frontend/redirection_interceptor.go [hunk 4]
```diff
@@ -177,8 +177,8 @@ func (i *RedirectionInterceptor) Intercept(
 		return i.handleRedirectAPIInvocation(ctx, req, info, handler, methodName, respAllocFn, namespaceName)
 	}
 
-	// this should not happen, log warn and move on
-	i.logger.Warn(fmt.Sprintf("RedirectionInterceptor encountered unknown API: %v", methodName))
+	// this should not happen for frontend APIs but ok for admin and other future handlers	
+	i.logger.Debug(fmt.Sprintf("RedirectionInterceptor encountered unknown API: %v", info.FullMethod))
```
Comment (dnr) [inline service/frontend/redirection_interceptor.go:181]: I'd still rather use tags.. this is going to actually fmt.Sprintf on every non-workflowservice rpc, right?

**Review by dnr — APPROVED** [review]

File: service/frontend/redirection_interceptor.go (hunk 4 above)
Comment (alexshtin) [reply service/frontend/redirection_interceptor.go:181]: yes, and not only for admin but for operator also

**Pull request merged**